javascript/javascript2/week1/SpiritNameGen.js

```diff
@@ -0,0 +1,34 @@
+const myButton = document.querySelector("button");
+let inputText;
+const spiritAnimalArray = [
+  "The Icy Bison",
+  "The Swift Armadillo",
+  "The Miniature Buffalo",
+  "The Capable Quail",
+  "The Gleaming Pig",
+  "The Enthusiastic Crane",
+  "The Amusing Penguin",
+  "The Sparkling Reindeer",
+  "The Expensive Caterpillar",
+  "The Intelligent Pony",
+];
+
+myButton.onclick = function (event) {
+  if (inputText.length > 0) {
```

Comment:
In here your function is breaking and it shows error "cannot read length property of undefined" when I click the button without filling in the name. 

This is because the program reads your if statement from the top and even though you have else statement where you account for the empty input possibility, it first starts by evaluating the if statement above it and it breaks ... To fix it you could think first checking whether inputText is an empty string (not to try to check its length or other properties which undefined objects don't have) or just check whether inputText returns a truthy value .

this is how your refactored function could look like
```
const section = document.querySelector("section"); // you will be using these couple of times so it's better to have access to them and save some typing ♥
  const p = document.createElement("p");
  const randomNumber = Math.floor(Math.random() * spiritAnimalArray.length);

myButton.onclick = function () {
  
  if (inputText) { // it returns true if inputText is truthy value (not nill or undefined)
    p.innerText = inputText + " - " + spiritAnimalArray[randomNumber];
  } else { // if  there is no input it will go on and execute your else statement
    p.innerText = "Please input your name to generate your spirit animal.";
  }

  section.appendChild(p); // also to avoid repeating the same code you can take the code you are executing in both statements out of them as the p.innerText will be different in each case
};
```